src/Microsoft.Framework.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -0,0 +1,478 @@
+        /// update where the expiration time is again extended by 60 mins.
+        /// </summary>
+        internal const int ExpirationTimeMultiplier = 2;
+
+        // Table column names
+        private const string CacheItemId = "Id";
+        private const string CacheItemValue = "Value";
+        private const string ExpiresAtTimeUTC = "ExpiresAtTimeUTC";
+        private const string SlidingExpirationInTicks = "SlidingExpirationInTicks";
+        private const string AbsoluteExpiration = "AbsoluteExpiration";
+        private int _cacheItemIdIndex = -1;
+        private int _cacheItemValueIndex = -1;
+        private int _expiresAtTimeUTCIndex = -1;
+        private int _slidingExpirationInTicksIndex = -1;
+        private int _absoluteExpirationIndex = -1;
+
+        private readonly SqlServerCacheOptions _options;
+        private readonly ILogger _logger;
+        private readonly ISystemClock _systemClock;
+        private readonly SqlQueries _sqlQueries;
+        private DateTimeOffset _lastExpirationScan;
+
+        public SqlServerCache(IOptions<SqlServerCacheOptions> options, ILoggerFactory loggerFactory)
+        {
+            _options = options.Options;
+            _systemClock = _options.SystemClock;
+            _sqlQueries = new SqlQueries(_options.TableName);
+            _logger = loggerFactory.CreateLogger<SqlServerCache>();
+        }
+
+        public void Connect()
+        {
+            // Try connecting to the database to test if its available and also
+            // get and cache the ordinal positions of the columns for better performance.
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.GetTableSchema, connection);
+                connection.Open();
+
+                var reader = command.ExecuteReader(CommandBehavior.SchemaOnly);
+
+                _cacheItemIdIndex = reader.GetOrdinal(CacheItemId);
+                _cacheItemValueIndex = reader.GetOrdinal(CacheItemValue);
+                _expiresAtTimeUTCIndex = reader.GetOrdinal(ExpiresAtTimeUTC);
+                _slidingExpirationInTicksIndex = reader.GetOrdinal(SlidingExpirationInTicks);
+                _absoluteExpirationIndex = reader.GetOrdinal(AbsoluteExpiration);
+            }
+        }
+
+        public async Task ConnectAsync()
+        {
+            // Try connecting to the database to test if its available and also
+            // get and cache the ordinal positions of the columns for better performance.
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.GetTableSchema, connection);
+                await connection.OpenAsync();
+
+                var reader = await command.ExecuteReaderAsync(CommandBehavior.SchemaOnly);
```

Comment:
Consider specifying sequential access to avoid overhead of filling whole row into memory, in particular in the async path this could be important (the only way to know is testing the perf :smile:). See http://blogs.msdn.com/b/adonet/archive/2012/07/15/using-sqldatareader-s-new-async-methods-in-net-4-5-beta-part-2-examples.aspx.